### PR TITLE
[react-interactions] Add TabFocusContainer and TabbableScope UI components

### PR DESCRIPTION
**Pull request by trueadm**
This PR adds two experimental components to be used for internal focus management handling. Furthermore, this PR fixes a bug with scope component refs correctly detaching.

`TabFocusContainer` contains tabbable keyboard focus to within its children, where tabbing gets wrapped to the beginning.

`TabbableScope` collects host components that are keyboard tabbable so that `TabFocusContainer` can handle the tab navigation through said components.

### CODE REVIEW

**Comment by necolas** [comment]
FWIW I still don't like the "tab" nomenclature given that these nodes can be reached by means other than pressing the Tab key on a keyboard.

**Comment by trueadm** [comment]
@necolas Maybe it should be better named as `TabContain` or something then, as this is exclusively for that operation only.

**Comment by necolas** [comment]
> this is exclusively for that operation only

so this component wouldn't work with e.g. a directional pad on a remote control?

File: packages/react-dom/src/client/focus/__tests__/TabFocusContainer-test.internal.js [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+let React;
+let ReactFeatureFlags;
+let TabFocusContainer;
+
+const createTabForward = type => {
```
Comment (necolas) [inline packages/react-dom/src/client/focus/__tests__/TabFocusContainer-test.internal.js]: You could use the testing-library helpers for this

File: packages/react-dom/src/client/focus/__tests__/TabFocusContainer-test.internal.js [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+          <button ref={buttonRef} />
+          <div ref={divRef} tabIndex={0} />
+          <input ref={input2Ref} tabIndex={-1} />
+          <button ref={butto2nRef} />
+        </TabFocusContainer>
+      );
+
+      ReactDOM.render(<Test />, container);
+      inputRef.current.focus();
+      document.activeElement.dispatchEvent(createTabForward());
```
Comment (necolas) [inline packages/react-dom/src/client/focus/__tests__/TabFocusContainer-test.internal.js]: You could use the testing-library helpers for this

Comment (trueadm) [reply packages/react-dom/src/client/focus/__tests__/TabFocusContainer-test.internal.js]: Sure. This was mostly a copy and paste from the old focus scope tests

**Pull request merged**